### PR TITLE
Use AD in initialization when the DAE algorithm uses AD

### PR DESCRIPTION
**Pull request by YingboMa**

### CODE REVIEW

File: src/initialize_dae.jl [hunk 1]
```diff
@@ -136,13 +136,14 @@ function _initialize_dae!(integrator, prob::ODEProblem, alg::ShampineCollocation
     nlequation! = function (out,u)
       update_coefficients!(M,u,p,t)
       #M * (u-u0)/dt - f(u,p,t)
-      @. tmp = (u - u0)/dt
+      tmp = @. (u - u0)/dt
```
Comment (ChrisRackauckas) [inline src/initialize_dae.jl:139]: why?

Comment (ChrisRackauckas) [reply src/initialize_dae.jl:139]: oh, the dual numbers. Can we make a DualCache here?

File: src/initialize_dae.jl [hunk 2]
```diff
@@ -287,12 +288,15 @@ function _initialize_dae!(integrator, prob::ODEProblem,
 	alg_u = @view u[algebraic_vars]
 
 	nlequation = (out, x) -> begin
-		alg_u .= x
-		f(tmp, u, p, t)
-		out .= @view tmp[algebraic_eqs]
+    uu = similar(x, axes(integrator.u)); copyto!(uu, integrator.u)
```
Comment (ChrisRackauckas) [inline src/initialize_dae.jl:291]: whitespace is odd here.

Comment (YingboMa) [reply src/initialize_dae.jl:291]: This file is all using tabs for some reason. I used spaces.

File: src/initialize_dae.jl (hunk 1 above)
Comment (YingboMa) [reply src/initialize_dae.jl:139]: I think DualCache is not compatible with arbitrary array length.

**Pull request merged**